jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ServletContextHandler.java

```diff
@@ -1195,7 +1196,23 @@ protected boolean handleByContextHandler(String pathInContext, ContextRequest re
         if (!initialDispatch)
             return false;
 
-        return super.handleByContextHandler(pathInContext, request, response, callback);
+        if (isProtectedTarget(pathInContext))
+        {
+            if (getErrorHandler() instanceof ErrorHandler.ErrorPageMapper mapper)
+            {
+                ErrorHandler.ErrorPageMapper.ErrorPage errorPage = mapper.getErrorPage(404, null);
+                if (errorPage != null)
+                {
+                    // Do nothing here other than set the error status so that the ServletHandler will handle as if a sendError
+                    request.setAttribute(ERROR_STATUS, 404);
+                    response.setStatus(HttpStatus.NOT_FOUND_404);
+                    return false;
+                }
+            }
+            Response.writeError(request, response, callback, HttpStatus.NOT_FOUND_404, null);
```

Comment:
I don't see how this ever sets the error attributes, as it doesn't go through the servlet `sendError`.

I actually think this doesn't work at all if they have a `org.eclipse.jetty.ee11.servlet.ErrorHandler` and want to handle the request with servlet APIs, because we never enter the `ServletChannel`.

Reply:
The `ErrorPageTest` in ee9,10,11 have tests that check that a configured error servlet is invoked, see: https://github.com/jetty/jetty.project/blob/jetty-12.1.x-12505-server-to-servlet-errorhandling/jetty-ee11/jetty-ee11-servlet/src/test/java/org/eclipse/jetty/ee11/servlet/ErrorPageTest.java#L2041

Reply:
@janbartel that test is only ever testing with `ErrorPageErrorHandler` so it is never reaching this line of code.

Reply:
@lachlan-roberts not following you - if they do not have an error page, but instead have configured a customized `ee11 ErrorHandler` or a `core ErrorHandler` then they're not using the servlet api, they're using jetty core api?

Reply:
I just assumed if you use the ee11 `ErrorHandler` class, you would be allowed to use the Servlet API if you override it.

But I guess that's not possible because we're returning true here, so they would have to use core APIs.

I don't understand why the EE11 does not combine `ErrorHandler`, `ErrorPageErrorHandler` and `ErrorPageMapper` into one class.